cmd/thanos/downsample.go

```diff
@@ -51,6 +53,11 @@ func newDownsampleMetrics(reg *prometheus.Registry) *DownsampleMetrics {
 		Name: "thanos_compact_downsample_failures_total",
 		Help: "Total number of failed downsampling attempts.",
 	}, []string{"group"})
+	m.downsampleDuration = promauto.With(reg).NewHistogramVec(prometheus.HistogramOpts{
+		Name:    "thanos_compact_downsample_duration_seconds",
+		Help:    "Duration of downsample runs",
+		Buckets: []float64{1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 2048, 4096, 8192},
+	}, []string{"resolution"})
```

Comment:
We can just reuse the label `group` to be aligned with other downsample metrics. This reflect the block resolution as well

Reply:
Makes sense. Could also help you identify the biggest offending group as well.